### PR TITLE
Revise DataVolumeTemplate API field

### PR DESCRIPTION
**Pull request by davidvossel**
The vm.Spec.DataVolumeTemplate api list actually contains DataVolumes rather than a struct that restricts the fields to only the metadata and spec. This gives the user the impression they can set things like the DataVolume group/kind and status when they in fact cannot. 

This change aligns our DataVolumeTemplate usage similar to how the PodTemplate works, by creating a new struct that only contains values a user can set.

I discovered this issue while reviewing this pr #3257 which adds the cdr validation schema back to our crd registrations. We're advertising DataVolume fields that users can't actually set. 

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by davidvossel** [comment]
/cc @xpivarc 

This will likely help us handle that strange datavolume exception logic you had to add to your explain pr.

File: staging/src/kubevirt.io/client-go/api/v1/types.go [hunk 1]
```diff
@@ -716,6 +716,15 @@ const (
 	VirtualMachineInstanceReplicaSetReplicaPaused VirtualMachineInstanceReplicaSetConditionType = "ReplicaPaused"
 )
 
+//
+// +k8s:openapi-gen=true
+type DataVolumeTemplateSpec struct {
+	// +nullable
+	metav1.ObjectMeta `json:"metadata,omitempty"`
```
Comment (xpivarc) [inline staging/src/kubevirt.io/client-go/api/v1/types.go:723]: This needs to be added: `// +kubebuilder:pruning:PreserveUnknownFields`

**Review by xpivarc — CHANGES_REQUESTED** [review]
I agree with this but please see my  https://github.com/kubevirt/kubevirt/pull/3257#discussion_r494802331.

**Comment by rmohr** [comment]
/override pull-kubevirt-e2e-k8s-1.17

Error related to #4264, overriding.

/lgtm
/approve

**Pull request merged**